eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Models/RequestPathPattern.cs

```diff
@@ -98,18 +94,42 @@ private bool TryTrimAncestorFrom(RequestPathPattern other, [MaybeNullWhen(false)
             }
             if (IsAncestorOf(other))
             {
-                diff = new RequestPathPattern(string.Join("", other._segments.Skip(Count)));
+                diff = new RequestPathPattern(other._segments.Skip(Count));
                 return true;
             }
             // Handle the special case of trim provider from feature
             else if (_path == ProviderPath && other._path.StartsWith(FeaturePath))
             {
-                diff = new RequestPathPattern(string.Join("", other._segments.Skip(Count + 2)));
+                diff = new RequestPathPattern(other._segments.Skip(Count + 2));
                 return true;
             }
             return false;
         }
 
+        public RequestPathPattern GetParent()
+        {
+            // if the request path has 0 or 1 segment, we call its parent the Tenant.
+            if (Count < 2)
+            {
+                return Tenant;
+            }
+
+            // otherwise, we try to remove its last two segments
+            if (Count < 4)
+            {
+                // if there are no more 4 segments, we trim off the last two segments as its parent
+                return new RequestPathPattern(_segments.Take(Count - 2));
+            }
+            // if there are 4 or more segments, we need to check if we trim off the last two segments, we get a `providers` segment pair
```

Comment:
```suggestion
            // if there are 4 or more segments, we need to check if we trim off the last 4 segments, we get a `providers` segment pair
```

Reply:
well this is not what I meant.
This is saying: when we have enough segments, we need to see if we really trim off 2 segment, we would get the `providers` pair, and then we need to trim 2 more.
Let me clarify